src/Model/Query/ReverseQuery.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Model\Query;
+
+use Geocoder\Model\Coordinates;
+use Geocoder\Provider\Provider;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+final class ReverseQuery
+{
+    /**
+     * @var Coordinates
+     */
+    private $coordinates;
+
+    /**
+     * @var int
+     */
+    private $limit;
+
+    /**
+     * @var string
+     */
+    private $locale;
+
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @param Coordinates $coordinates
+     */
+    private function __construct(Coordinates $coordinates)
+    {
+        $this->coordinates = $coordinates;
+        $this->data = [];
+        $this->limit = Provider::MAX_RESULTS;
+    }
+
+    /**
+     * @param Coordinates $coordinates
+     *
+     * @return ReverseQuery
+     */
+    public static function create(Coordinates $coordinates)
+    {
+        return new self($coordinates);
+    }
+    /**
+     * @param float $latitude
+     * @param float $longitude
+     *
+     * @return ReverseQuery
+     */
+    public static function fromCoordinates($latitude, $longitude)
```

Comment:
Personally I find this a bit more convenience stuff than necessary. Either create or fromCoordinates IMHO.